lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -9,7 +9,8 @@ class Base
           ::Errno::ENOENT,
           ::Racc::ParseError,
           ::RubyParser::SyntaxError,
-        ]
+          ::RuntimeError,
+        ].freeze
```

Comment:
One important case we *don't* want to rescue is timeouts parsing individual files: we found that previously the timeout values were set low enough that GC-jitter/system pressure/etc. could cause a file to timeout on some runs & not others (leading to inconsistent results). So the timeout values have been greatly heightened to the point where they indicate real problems, and we don't want to catch them because they indicate a file should be excluded/there's a more severe parser bug that deserves investigation. The problems caused by this is why we moved away from the prior behavior, which did catch basically all exceptions.

Beyond that, I'm still of the opinion that exceptions can & normally should indicate bugs that need fixing, and when they're caught/handled we should be as precise as possible about what we're catching. Otherwise we're potentially just hiding bugs. In this particular case we're intentionally hiding bugs in a dependency (`RubyParser`, etc.), which isn't great, but seems like the lesser of the evils available to us.